### ISSUE TITLE
ToolBarItem is increasing with switch the tabs

### ISSUE THREAD

**Issue by nomurayoshuak** [adospace/reactorui-maui#261]
@adospace 
I really really appreciate for the you fixed the #257.
But I face the new bug `ToolBarItem` is increasing with switch the tabs that in the same situation of the #257.
Here is my `Render()` and the screenshots.
Do you fix that?

```c#
public override VisualNode Render()
{
        return ContentPage(
            ToolbarItem()
                .IconImageSource(new MauiControls.FontImageSource() {FontFamily = "material", Glyph = "light_mode", Color = AppTheme.Scheme.OnSurface})
                .Order(MauiControls.ToolbarItemOrder.Primary),
                
            ToolbarItem()
                .IconImageSource(new MauiControls.FontImageSource() {FontFamily = "material", Glyph = "refresh", Color = AppTheme.Scheme.OnSurface})
                .Order(MauiControls.ToolbarItemOrder.Primary),
            
            // other components
}
```

first time (no switching tab, no problem)
![Screenshot_first_view](https://github.com/user-attachments/assets/0cfdd1bb-d129-4d90-a94d-fa3b90080395)

second time (once switching tab, ToolBarItem is increased)
![Screenshot_second_time_view](https://github.com/user-attachments/assets/84d3b3dc-13d6-42c6-a180-9a27ec370151)

third time (twice switching tab, ToolBarItem is increased again)
![Screenshot_third_time_view](https://github.com/user-attachments/assets/8eb5f233-62cb-49d9-aeaa-003cfab54526)

**Comment by nomurayoshuak** [comment]
@adospace 
Thank you for your quick response.
I've never done testing in a specific branch before, but I'll give it a try.
Is it correct way to reference the MauiReactor project and MauiReactor.Canvas project of a specific branch instead of referencing Nuget's ones?

**Comment by adospace** [comment]
Yes, anyway do you see any other problem with this version? from testing, it seems this is the only one.
Please let me know if you find anything else.

**Comment by adospace** [comment]
Yes the main idea is to keep the MauiReactor behavior as close as possible to the native MAUI library.
The main problem is that it's hard to understand when the page is definitely unloaded. 
For example, in your case MAUI raise a Disappearing event when you switch to another tab and Appearing when you move back to Page6: Page6 is NOT unloaded so the component should be kept mounted.
But if you just go back to Home tab from Page6 raise the Disappearing event AND the page is unloaded so the component should be unmounted BUT MAUI doesn't raise an event to signal that page is definitively gone.
This is causing some troubles with MauiReactor to identify when components should be mounted/unmounted.
I'm still working on resolve the issue considering your use case.

**Comment by adospace** [comment]
@nomurayoshuak I've published an update on the branch that fixes other small wrong behaviors.
Regarding your case, unfortunately, MAUI sends a disappearing event for the Page6 while the NavigationStack doesn't contain it and this signals MauiReactor that the component must be unmounted.
In any other case, MauiReactor correctly leaves the component live because the page is still present in the NavigationStack.
To sum up for now, in your specific case, (maybe in .net 9 things will change), MauiReactor has no other choice than unmounting the component when you move to another tab.
Please check if your app still works fine using this branch soon I'll publish the new package version.

**Comment by nomurayoshuak** [comment]
Thank you. I understand it.
Although I may not be able to help you, I wish you and this library success.
MauiReactor is great library.
I very appreciate.

**Comment by nomurayoshuak** [comment]
@adospace 
I checked the recent `fix-261` branch with my app. I works fine! thank you!
